maths/sphenic_number.dart

```diff
@@ -0,0 +1,50 @@
+var arr = new List<int>.filled(1001, 1, growable: false);
+void simple_seive() {
+  var i;
+
+  var p;
+  for (p = 2; p * p < 1001; p++) {
+    if (arr[p] == 1) {
+      for (i = p * 2; i < 1001; i = i + p) arr[i] = 0;
+    }
+  }
```

Comment:
You could initialize `i` and `p` in for loop
```suggestion
  for (int p = 2; p * p < 1001; p++) {
    if (arr[p] == 1) {
      for (int i = p * 2; i < 1001; i = i + p) arr[i] = 0;
    }
  }
```